src/sugar3/graphics/progressicon.py

```diff
@@ -14,86 +14,154 @@
         cr.translate(margin_x, margin_y)
 
         # Paint the fill, clipping it by the progress.
-        x_, y_ = 0, 0
-        width, height = self._stroke.get_width(), self._stroke.get_height()
-        if self._direction == 'vertical':  # vertical direction, bottom to top
+        x_, y_ = 0, 0 '''The value of margin x_ and y_ is 0,0 (0i,0j) '''
+        width, height = self._stroke.get_width(), self._stroke.get_height() '''Width = self._stroke.get_width() & height = self._stroke.get_height()'''
+        if self._direction == 'vertical':  '''if direction is vertical then...''' #vertical direction, bottom to top 
             y_ = self._stroke.get_height()
             height *= self._progress * -1
-        else:
+        else: 
             rtl_direction = \
-                Gtk.Widget.get_default_direction() == Gtk.TextDirection.RTL
-            if rtl_direction:  # horizontal direction, right to left
-                x_ = self._stroke.get_width()
+                Gtk.Widget.get_default_direction() == Gtk.TextDirection.RTL '''Sets text direction from right to left'''
+            if rtl_direction: '''If rtl_direction ----> right to left '''  # horizontal direction, right to left
+                x_ = self._stroke.get_width() '''value of x_ gets set as self._stroke.get_width() '''
                 width *= self._progress * -1
-            else:  # horizontal direction, left to right
+            else: '''IF rtl_direction ----> left to right '''  # horizontal direction, left to right
                 width *= self._progress
 
-        cr.rectangle(x_, y_, width, height)
-        cr.clip()
+        cr.rectangle(x_, y_, width, height) '''x_ = width and y_ = width '''
+        cr.clip() 
         cr.set_source_surface(self._fill, 0, 0)
         cr.paint()
 
         # Paint the stroke over the fill.
-        cr.reset_clip()
+        cr.reset_clip() 
         cr.set_source_surface(self._stroke, 0, 0)
         cr.paint()
 
     def do_get_preferred_width(self):
-        width = self._stroke.get_width()
-        return (width, width)
+        width = self._stroke.get_width() 
+        return (width, width) 
+        '''Returns tuple'''
+        '''
+        Example:
+
+        def do_get_preferred_width(self):
+        print("example")
+        return (100, 100)
+        
+        '''
 
     def do_get_preferred_height(self):
-        height = self._stroke.get_height()
-        return (height, height)
-
-    def update(self, progress):
-        self._progress = progress
-        self.queue_draw()
+        height = self._stroke.get_height() '''height will be set as self._stroke.get_height()'''
+        return (height, height) '''Returns tuple'''
+        '''
+        Example:
+        
+        def do_get_preferred_height(self):
+        print("example")
+        return (100, 100)
+        
+        '''
+
+    def update(self, progress): '''updates progressicon with progress value '''
```

Comment:
Please document this function in a way simmilar to the example (where the doc string is directly under the function def)